src/Fmt_ast.ml

```diff
@@ -1759,11 +1771,16 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
                , fmt_after_cmts
                , (fmt_op, [(Nolabel, arg)]) ))) )
   | Pexp_construct (({txt= Lident "::"; loc= _} as lid), Some arg) ->
+      let opn, cls =
+        let ( ! ) = Caml.format_of_string in
```

Comment:
Minor: Instead you could partially apply `wrap` to force the type without annotation and introducing a new syntax.

Reply:
Fixed.